### ISSUE TITLE
summary statistics (mean, max, min, etc.)

### ISSUE THREAD

**Issue by pdurbin** [mlcommons/croissant#640]
As mentioned during the 2024-04-10 task force call, there is interest in providing summary statistics (or more broadly, descriptive statistics) in Croissant format. I'm focusing on summary statistics (mean, max, min, median, mode, standard deviation, etc.) because they are already well defined in the [Data Document Initiative][] (DDI) format.

[Data Document Initiative]: https://en.wikipedia.org/wiki/Data_Documentation_Initiative

For example, for [a dataset][] with a variable called "stars" that indicates the number of stars on GitHub, the summary statistics in DDI can be represented like this:

[a dataset]: https://doi.org/10.7910/DVN/TJCLKP

```
<var ID="v30256083" name="stars" intrvl="discrete">
  <location fileid="f6867331"/>
  <labl level="variable">stars</labl>
  <sumStat type="medn">4.0</sumStat>
  <sumStat type="mean">38.71014492753635</sumStat>
  <sumStat type="mode">.</sumStat>
  <sumStat type="vald">138.0</sumStat>
  <sumStat type="max">732.0</sumStat>
  <sumStat type="invd">0.0</sumStat>
  <sumStat type="min">0.0</sumStat>
  <sumStat type="stdev">110.13079171235681</sumStat>
  <varFormat type="numeric"/>
</var>
```

The question is, where can I put summary statistics in Croissant?

**Comment by B13rg** [comment]
I agree with the use of DDI formats, they seem pretty well thought out and designed.
Based on [Controlled Vocaularies](https://rdf-vocabulary.ddialliance.org/), there are a few "Controlled Vocabulary" (CV) definitions that may be useful:

* [SummaryStatisticType](https://rdf-vocabulary.ddialliance.org/ddi-cv/SummaryStatisticType/2.1.2/SummaryStatisticType.html) of course
* [AggregationMethod](https://rdf-vocabulary.ddialliance.org/ddi-cv/AggregationMethod/1.1.2/AggregationMethod.html)
* [TypeOfTranslationMethod](https://rdf-vocabulary.ddialliance.org/ddi-cv/TypeOfTranslationMethod/1.0.0/TypeOfTranslationMethod.html)

While it would only be referenced, it may be a lot for implementations/tooling to support.
Instead of referencing the vocabulary maybe a partial list of fields could be added to keep things simpler.
On the other hand, it's probably better to be specific and thorough to prevent users from creating non-spec'd standards, especially with the plan of adding  of annotations #737 #739 .

It's probably easiest for the 1.1 spec to reference the DDI CV version numbers.
There are RDF representations [available](https://documenter.getpostman.com/view/2085804/2s8ZDVYiHx), so it might be valuable to use it to generate spec and lib definitions.

**Comment by jqnatividad** [comment]
We're currently working on inferring metadata from raw data while populating CKAN catalogs, mapping it to standards like Croissant and DCAT3 leveraging the work done in ckanext-dcat and ckanext-scheming.

https://ckan.org/blog/bridging-ckan-and-machine-learning-introducing-support-for-the-croissant-standard

We're able to do so with a data-wrangling tool we maintain called qsv. 

https://github.com/dathere/qsv

It can infer summary stats from a tabular file very quickly (e.g. for a 1m row, 520 mb, 41 column sample of NYC's 311 data, it can infer data type and 43 summary stats in 1.5 seconds).

https://github.com/dathere/qsv/wiki/Supplemental#stats-command-output-explanation

Further, for the same file, it can compile a frequency table in 0.9 seconds.

For DCAT-US v3, we're implementing an expanded, "validating" data dictionary with summary statistics and frequency table for its [describedBy](https://doi-do.github.io/dcat-us/#dataset-describedBy) property.  For this, we're pointing it to a JSONSchema inferred using the `qsv schema` command, making use of JSONSchema's `additionalProperties` to add non-standard JSON Schema properties.

Would be great if we can use the same technique to add Summary statistics and optionally, frequency tables, to Croissant.
